cdm/src/main/java/ucar/nc2/ft2/coverage/simpgeometry/SimpleGeometryFeatureDataset.java

```diff
@@ -0,0 +1,240 @@
+import ucar.nc2.time.CalendarDate;
+import ucar.nc2.time.CalendarDateRange;
+import ucar.nc2.util.cache.FileCacheIF;
+import ucar.unidata.geoloc.LatLonRect;
+
+import java.io.IOException;
+import java.util.*;
+
+public class SimpleGeometryFeatureDataset implements FeatureDataset {
+
+    private NetcdfDataset ncd;
+
+    /**
+     * Open a netcdf dataset, using NetcdfDataset.defaultEnhanceMode plus CoordSystems
+     * and turn into a SimpleGeometryFeatureDataset.
+     *
+     * @param location netcdf dataset to open, using NetcdfDataset.acquireDataset().
+     * @return SimpleGeometryFeatureDataset
+     * @throws java.io.IOException on read error
+     * @see ucar.nc2.dataset.NetcdfDataset#acquireDataset
+     */
+    static public SimpleGeometryFeatureDataset open(String location) throws IOException {
+        return open(location, NetcdfDataset.getDefaultEnhanceMode());
+    }
+
+    /**
+     * Open a netcdf dataset, using NetcdfDataset.defaultEnhanceMode plus CoordSystems
+     * and turn into a SimpleGeometryFeatureDataset.
+     *
+     * @param location netcdf dataset to open, using NetcdfDataset.acquireDataset().
+     * @param enhanceMode open netcdf dataset with this enhanceMode
+     * @return SimpleGeometryFeatureDataset
+     * @throws java.io.IOException on read error
+     * @see ucar.nc2.dataset.NetcdfDataset#acquireDataset
+     */
+    static public SimpleGeometryFeatureDataset open(String location, Set<NetcdfDataset.Enhance> enhanceMode) throws IOException {
+        NetcdfDataset ds = ucar.nc2.dataset.NetcdfDataset.acquireDataset(null, DatasetUrl.findDatasetUrl(location), enhanceMode, -1, null, null);
+        return new SimpleGeometryFeatureDataset(ds);
+    }
+
+    /**
+     * Create a SimpleGeometryFeatureDataset from a NetcdfDataset.
+     *
+     * @param ncd underlying NetcdfDataset, will do Enhance.CoordSystems if not already done.
+     * @throws java.io.IOException on read error
+     */
+    public SimpleGeometryFeatureDataset(NetcdfDataset ncd) throws IOException {
+        this.ncd = ncd;
+        Set<NetcdfDataset.Enhance> enhance = ncd.getEnhanceMode();
+        if(enhance == null || enhance.isEmpty()) enhance = NetcdfDataset.getDefaultEnhanceMode();
+        ncd.enhance(enhance);
+    }
+
+    // stuff to satisfy ucar.nc2.dt.TypedDataset
+    public String getTitle() {
+        String title = ncd.getTitle();
+        if (title == null)
+            title = ncd.findAttValueIgnoreCase(null, CDM.TITLE, null);
+        if (title == null)
+            title = getName();
+        return title;
+    }
+
+    public String getDescription() {
+        String desc = ncd.findAttValueIgnoreCase(null, "description", null);
+        if (desc == null)
+            desc = ncd.findAttValueIgnoreCase(null, CDM.HISTORY, null);
+        return (desc == null) ? getName() : desc;
+    }
+
+    public String getLocation() {
+        return (ncd != null) ?  ncd.getLocation() : "";
+    }
+

```

Comment:
Why do the following batch o' methods return `null`? Is it within scope to implement some of them?

Reply:
Is there any prior art implementing `FeatureDataset` that we could draw inspiration from?

Reply:
as discussed, postponing until we know more